### PR TITLE
GLTFLoader: Callback for when asset was read

### PR DESCRIPTION
**Pull request by cnspaha**
Currently there is absolutely no possibility to access the asset property from the loaded GLTF. 
This here is one idea how it could look like. It is similar like "onAfterRender" etc where the developer just have to hook on the function to get the value.

Opinions regarding naming etc are welcome.
Also should we rather call this callback right after checking for the supported version or before?  The downside of the current code is that your callback won't be called if the GLTF version isn't supported, but then you normally won't have any need for the information in ```assets``...

### CODE REVIEW

**Comment by cnspaha** [comment]
An other possibility (and probably the better) is to put the ```asset``` into the returned object
```javascript
parser.parse( function ( scene, scenes, cameras, animations, asset ) { // also set 'asset'

  console.timeEnd( 'GLTFLoader' );

  var glTF = {
    scene: scene,
    scenes: scenes,
    cameras: cameras,
    animations: animations,
    asset: asset  // this line is new
  };

  onLoad( glTF );
}, onError );

```

**Comment by takahirox** [comment]
Voting for `onLoad` argument.

**Comment by donmccurdy** [comment]
Curious why this is needed? We're hoping to make the loader more extensible in the future; see #11682; I imagine we would revert this change when that happens..

Of the two approaches I do prefer including a property on the returned object, though.

**Comment by donmccurdy** [comment]
thinking more about it, I'm happy with the `onLoad` `asset` argument idea. Could imagine returning `materials` that way too eventually, so that it's possible to use a glTF file as a material library.

**Comment by cnspaha** [comment]
@donmccurdy 
just rewrote it to the ```onLoad asset``` variety. please have a look ;)

**Review by donmccurdy — APPROVED** [review]

**Comment by donmccurdy** [comment]
@cnspaha there has been an upstream change, could you rebase this branch?

**Comment by cnspaha** [comment]
@donmccurdy 
sorry just didn't accomplished it to rebase this... don't know why git didn't wanted to :/ 
So I merged down the dev down to my patch-3 branch to resolve the conflicts with the base

**Comment by donmccurdy** [comment]
No problem, this looks good. 👍

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -1241,9 +1242,10 @@ THREE.GLTFLoader = ( function () {
 			var scenes = dependencies.scenes || [];
 			var scene = scenes[ json.scene || 0 ];
 			var animations = dependencies.animations || [];
+			var asset = json.asset;
 			var cameras = dependencies.cameras || [];
 
-			onLoad( scene, scenes, cameras, animations );
+			onLoad( scene, scenes, cameras, animations, asset );
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: What about passing `json` directly?

Comment (cnspaha) [reply examples/js/loaders/GLTFLoader.js]: would be possible of course, but then we would pass the whole GLTF data back.
In my opinion this is a bit of overkill so I would stick with the ```asset```. Or are there any other values in the GLTF which would be good to know as caller of the loader? I am not so familiar with the structure of the GLTF format

**Comment by mrdoob** [comment]
I see. Well, if @donmccurdy is cool with this I'm cool with this 😊

**Comment by mrdoob** [comment]
Thanks!

**Comment by donmccurdy** [comment]
> What about passing json directly?

This might be worth considering; added a comment on https://github.com/mrdoob/three.js/issues/11682 to revisit later.

**Pull request merged**